redfish-core/lib/log_services.hpp

```diff
@@ -841,6 +842,21 @@ inline void deleteDumpEntry(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                 messages::resourceNotFound(asyncResp->res, "LogEntry", entryID);
                 return;
             }
+
+            const sd_bus_error* dbusError = msg.get_error();
+            if (dbusError == nullptr)
+            {
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            if (strcmp(dbusError->name, "xyz.openbmc_project.Common.Error."
```

Comment:
Remove line number 846 till 852. You can check if (dbusError!=nullptr && dbusError == NotAllowed) as single statement

Reply:
if (dbusError == nullptr) then we need to internal error.
it hits this check at 853 when dbusError != nullptr